rmb-sdk-go/peer/connection.go

```diff
@@ -133,10 +140,12 @@ func (c *InnerConnection) loop(ctx context.Context, con *websocket.Conn, output
 		case sent := <-c.writer:
 			err := con.WriteMessage(websocket.BinaryMessage, sent.data)
 
-			select {
-			case sent.err <- err:
-			case <-ctx.Done():
-				return ctx.Err()
+			if !sent.closed.Load() {
+				select {
+				case sent.err <- err:
+				case <-ctx.Done():
+					return ctx.Err()
+				}
```

Comment:
Still potential race condition here. Even if u use atomic bool, what prevents us from seeing that closed is false, then later the send is closed while blocked on the write inside the select statement

Proposed solution is the following:
- Implement a `reply(ctx Context, err error)` function on the send object
- replace this block with a call `send.reply(ctx, err)`
- in the reply implementation you still do 
```
select {
  case <-ctx.Done():
     return ctx.Err()
  case send.err <- err:
```
But you also add a defer recover section.

Any check you gonna make will have a potential race condition anyhow

Reply:
Can be better if we include the `send` context from the send itself. Since the send context can be different from the context passed to the loop. You will have to try this out 

so basically  in send
```
send := send {
   ctx: ctx
   err: (channel)
   data: ...
}
```

 in the reply(err error)
```
select {
  // <- self.ctx 
  // <- write error
```